credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -246,6 +246,8 @@ func (h *altsHandshaker) ServerHandshake(ctx context.Context) (net.Conn, credent
 	if err != nil {
 		return nil, nil, err
 	}
+	inBytes := make([]byte, n)
+	copy(inBytes, p)
```

Comment:
The way it currently stands, `p` and `p[:n]` are two different slices with the same backing byte array which was allocated with the call to `make([]byte, frameLimit)`. Once this function returns both slices should go out of scope and hence the backing array should get GC'ed.

Am I missing something? Could you maybe explain why the buffer is leaked in the existing implementation.
Thanks.

Reply:
The backing byte array will keep being referenced in the conn returned by 
`conn, result, err := h.doHandshake(req)`

Reply:
Why would the `conn` hold a reference to the `req` (which is what holds the reference to the backing array)?

Reply:
conn does not hold a reference to req, but holds the reference to p array. If you trace down the doHandshake function, the p array was transformed into a extra slice and is referenced in the protected field  of the conn object.

Reply:
I think I understand it now. Thanks.

Reply:
NP